src/AzureSignTool/SignCommand.cs

```diff
@@ -294,6 +312,33 @@ public async Task<int> OnExecuteAsync(CommandLineApplication app, IConsole conso
                         {
                             return state;
                         }
+
+                        if (SigningThrottle is > 0)
+                        {
+                            DateTime goTime = _lastSigningOperation.AddSeconds(SigningThrottle.Value);
+
+                            if (goTime >= DateTime.Now)
```

Comment:
I have some concerns around this since DateTime.Now is not monotonic - for example, a Daylight Saving Time. Let's say the `_lastSigningOperation` is at 1:59 AM. Day Light Savings kicks in at 2:00 AM, and jumps forward to 3:00 AM. This will violate the throttle.

Using `DateTimeOffset.UtcNow` probably mitigates most of those concerns... except for things like clock correction. I think that is a tradeoff I am willing to accept. A "true" monotonic clock source is probably overkill for this purpose.

Can we re-work this to use DateTimeOffset.UtcNow?

Reply:
Could possibly use System.Threading.PeriodicTimer instead, which was added at some point, but I'd need to check the version availability for it. You just call the wait method on the class, it'll return immediately if there's already been one or more ticks since the method was last called, or it was wait until there is one then return.